internal/settings.go

```diff
@@ -70,8 +70,9 @@ type DialSettings struct {
 	RequestReason string
 
 	// New Auth library Options
-	AuthCredentials      *auth.Credentials
-	EnableNewAuthLibrary bool
+	AuthCredentials          *auth.Credentials
+	EnableNewAuthLibrary     bool
+	EnableAsyncRefreshDryRun func()
```

Comment:
Should all of the changes in this PR be marked with `TODO` comments explaining that they are temporary and should be removed after some event?